File: src/main/java/org/smudge/petstore/PersonController.java
package org.smudge.petstore;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("person")
public class PersonController {

    @Autowired
    private PersonService personService;

    @GetMapping()
    public Person getPerson(@RequestParam int id) {
        return personService.getPerson(id);
    }

    @PostMapping
    public Person createPerson(@RequestBody Person newPerson) {
        return personService.createPerson(newPerson.getName());
    }

    @PutMapping
    public Person changeName(@RequestParam int id, @RequestParam String newName) {
        return personService.changeName(id, newName);
    }
}

File: src/main/java/org/smudge/petstore/PersonController.kt
package org.smudge.petstore

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("person")
class PersonController(
        @Autowired
        private val personService: PersonService
) {

    @GetMapping
    fun getPerson(@RequestParam id: Int): Person {
        return personService.getPerson(id)
    }

    @PostMapping
    fun createPerson(@RequestBody newPerson: Person): Person {
        return personService.createPerson(newPerson.name)
    }

    @PutMapping
    fun changeName(@RequestParam id: Int, @RequestParam newName: String = "Bob"): Person {
        return personService.changeName(id, newName)
    }
}